spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/config/parameterstore/ParameterStoreConfigDataLoaderIntegrationTests.java

```diff
@@ -115,6 +134,21 @@ void ssmClientCanBeOverwrittenInBootstrapConfig() {
 		}
 	}
 
+	@Test
+	void checkIfClientIsConfiguredForTimeout() {
+		SpringApplication application = new SpringApplication(App.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+
+		try (ConfigurableApplicationContext context = runApplication(application,
+				"aws-parameterstore:/config/spring/")) {
+			SsmClient ssmClient = context.getBean(SsmClient.class);
+			// very ugly have to check if there is better way to access this field
+			Map attributeMap = (Map) ReflectionTestUtils.getField(ReflectionTestUtils.getField(
+					ReflectionTestUtils.getField(ssmClient, "clientConfiguration"), "attributes"), "attributes");
```

Comment:
I have not found better way to test that fields are being set besides used reflection.